### PR TITLE
`terminalBell` -> `terminalVisualBell` and `signals.terminalBell`

### LINKED ISSUE #205974: terminal bell not working
Does this issue occur when all extensions are disabled?: Yes

Version: 1.86.2 (user setup)
Commit: 903b1e9d8990623e3d7da1df3d33db3e42d80eda
Date: 2024-02-13T19:40:56.878Z
Electron: 27.2.3
ElectronBuildId: 26908389
Chromium: 118.0.5993.159
Node.js: 18.17.1
V8: 11.8.172.18-electron.0
OS: Windows_NT x64 10.0.19045

Steps to Reproduce:

1. enable terminal.integrated.enablebell
2. try to echo a bell command e.g. git bash echo -n $'\a'

this has worked before, some weeks ago, even with remote ssh repository. but since some weeks, the audio bell is just gone. i then tried it with my private pc and updated to the same vscode version my work pc has running and the bell did not work either. A colleague of mine also tried it and it didnt work on his workstation either. I tried the insiders version of vscode, but actually the whole audio cues settings are missing there. why would someone disable or remove the terminal bell feature? As far as i can see, the terminal.integrated.enablebell is changed to only enable the visual bell, but not the sound bell anymore. I used this to let me know when a very-long running build is finished and this is really handy as you can focus on other work until your main work can continue. please either fix the bug or re-insert the feature

### PR DESCRIPTION
**Pull request by meganrogge**
fix #205974

### CODE REVIEW

File: src/vs/workbench/contrib/terminal/common/terminalConfiguration.ts [hunk 1]
```diff
@@ -660,3 +666,20 @@ export function registerTerminalConfiguration() {
...
+	.registerConfigurationMigrations([{
+		key: TerminalSettingId.EnableBell,
+		migrateFn: (enableBell, accessor) => {
+			const configurationKeyValuePairs: ConfigurationKeyValuePairs = [];
+			configurationKeyValuePairs.push([TerminalSettingId.EnableBell, { value: undefined }]);
+			let announcement = accessor('accessibility.signals.terminalBell')?.announcement ?? accessor('accessibility.alert.terminalBell');
+			if (announcement !== undefined && typeof announcement !== 'string') {
+				announcement = announcement ? 'auto' : 'off';
+			}
+			configurationKeyValuePairs.push(['accessibility.signals.terminalBell', { value: { sound: enableBell ? 'on' : 'off', announcement } }]);
```
Comment (Tyriar) [inline src/vs/workbench/contrib/terminal/common/terminalConfiguration.ts:680]: I don't think we want to return anything if `enableBell` was not set?

File: src/vs/workbench/contrib/terminal/common/terminalConfiguration.ts [hunk 2]
```diff
@@ -660,3 +666,20 @@ export function registerTerminalConfiguration() {
 	const configurationRegistry = Registry.as<IConfigurationRegistry>(Extensions.Configuration);
 	configurationRegistry.registerConfiguration(terminalConfiguration);
 }
+
+Registry.as<IConfigurationMigrationRegistry>(WorkbenchExtensions.ConfigurationMigration)
+	.registerConfigurationMigrations([{
+		key: TerminalSettingId.EnableBell,
+		migrateFn: (enableBell, accessor) => {
+			const configurationKeyValuePairs: ConfigurationKeyValuePairs = [];
+			configurationKeyValuePairs.push([TerminalSettingId.EnableBell, { value: undefined }]);
```
Comment (Tyriar) [inline src/vs/workbench/contrib/terminal/common/terminalConfiguration.ts:675]: This is added twice to the array?

File: src/vs/workbench/contrib/terminal/common/terminalConfiguration.ts [hunk 3]
```diff
@@ -660,3 +666,20 @@ export function registerTerminalConfiguration() {
 	const configurationRegistry = Registry.as<IConfigurationRegistry>(Extensions.Configuration);
 	configurationRegistry.registerConfiguration(terminalConfiguration);
 }
+
+Registry.as<IConfigurationMigrationRegistry>(WorkbenchExtensions.ConfigurationMigration)
+	.registerConfigurationMigrations([{
+		key: TerminalSettingId.EnableBell,
+		migrateFn: (enableBell, accessor) => {
```
Comment (Tyriar) [inline src/vs/workbench/contrib/terminal/common/terminalConfiguration.ts:673]: My thinking was to not migrate it initially and add this a little later so users get a chance to understand the change/expanded functionality before we do it automatically.

**Review by Tyriar — CHANGES_REQUESTED** [review]

File: src/vs/workbench/contrib/terminal/common/terminalConfiguration.ts (hunk 2 above)
Comment (meganrogge) [reply src/vs/workbench/contrib/terminal/common/terminalConfiguration.ts:675]: not sure what you mean

File: src/vs/workbench/contrib/terminal/common/terminalConfiguration.ts (hunk 1 above)
Comment (meganrogge) [reply src/vs/workbench/contrib/terminal/common/terminalConfiguration.ts:680]: this function is only run if a user has set `enableBell`

File: src/vs/workbench/contrib/terminal/common/terminalConfiguration.ts (hunk 2 above)
Comment (Tyriar) [reply src/vs/workbench/contrib/terminal/common/terminalConfiguration.ts:675]: It's the same content on line 681

Comment (meganrogge) [reply src/vs/workbench/contrib/terminal/common/terminalConfiguration.ts:675]: 🤦🏼‍♀️  the rare case when viewing the diff in vscode vs on github makes things clearer.

**Review by Tyriar — APPROVED** [review]

**Pull request merged**